editoast/schemas/src/rolling_stock.rs

```diff
@@ -174,3 +175,77 @@ impl Serialize for RollingStock {
         RollingStock::serialize(self, serializer)
     }
 }
+
+impl RollingStock {
+    /// Returns the supported signaling systems, including "ETCS_LEVEL2" if `etcs_brake_params` is set.
+    pub fn get_all_supported_signaling_systems(&self) -> RollingStockSupportedSignalingSystems {
+        let mut supported_signaling_systems = self.supported_signaling_systems.0.clone();
+        if self.etcs_brake_params.is_some() {
+            supported_signaling_systems.push("ETCS_LEVEL2".into());
+        }
+        RollingStockSupportedSignalingSystems(supported_signaling_systems)
+    }
+
+    pub fn get_row_supported_signaling_systems(&self) -> RollingStockSupportedSignalingSystems {
+        self.supported_signaling_systems.clone()
+    }
+
+    pub fn set_row_supported_signaling_systems(
```

Comment:
Let's clarify the naming and make it available only for tests:
```suggestion
    #[cfg(test)]
    pub fn set_raw_supported_signaling_systems_unchecked(
```